### DISCUSSION TITLE
ruff analyze-graph to be inspired by tach?

### DISCUSSION THREAD

**Discussion by gwdekker** [astral-sh/ruff#13565] [Ideas]
https://github.com/gauge-sh/tach is an interesting tool that has provided a lot of value for me in its short existence. Is ruff analyze aiming to replace its feature set? I really appreciate in practice their idea of having a toml file that shows the current dependency structure of the repository, which then can be used to inspect and improve certain dependency rules - for example, use it to make sure certain architectural patterns are followed regarding which imports are allowed.

**Comment by romain-dartigues** [comment]
I'd also love to have an efficient way to list dependencies; until now I used https://github.com/thebjorn/pydeps, https://github.com/stefano-maggiolo/pydepgraph, https://github.com/tox-dev/pipdeptree... (and some hooks for Sphinx-doc).
